app/lib/common/widgets/scrollable_list_tab_scroller.dart

```diff
@@ -363,37 +365,51 @@ class DefaultHeaderWidget extends StatefulWidget {
 }
 
 class _DefaultHeaderWidgetState extends State<DefaultHeaderWidget>
-    with SingleTickerProviderStateMixin {
-  late final TabController _tabController;
+    with TickerProviderStateMixin {
+  TabController? _tabController;
 
   @override
   void initState() {
     super.initState();
+    _setController();
+  }
+
+  void _setController() {
+    _tabController?.dispose();
+    widget.selectedTabIndex.removeListener(externalTabChangeListener);
     _tabController = TabController(
       length: widget.itemCount,
       vsync: this,
     );
-    _tabController.addListener(tabChangeListener);
+    _tabController!.addListener(tabChangeListener);
     widget.selectedTabIndex.addListener(externalTabChangeListener);
   }
 
+  @override
+  void didUpdateWidget(covariant DefaultHeaderWidget oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.itemCount != widget.itemCount) {
+      _setController();
+    }
+  }
+
   @override
   void dispose() {
-    _tabController.dispose();
+    _tabController?.dispose();
```

Comment:
Yes, about here. Isn't this enough to dispose off controller ? Also I think this [argument](https://github.com/acterglobal/a3/pull/2077#discussion_r1724902072) was raised in my PR, this should be disposed off automatically, no?